src/peripheral/nvic.rs

```diff
@@ -33,9 +33,16 @@ pub struct RegisterBlock {
     /// Interrupt Active Bit (not present on Cortex-M0 variants)
     #[cfg(not(armv6m))]
     pub iabr: [RO<u32>; 16],
-    #[cfg(armv6m)]
+    #[cfg(any(armv6m, armv8m))]
```

Comment:
This change doesn't seem right to me - armv8m still has IABR right? And in any event the matching cfg above isn't changed, so armv8 would end up with an extra 16 byte offset here.

Reply:
This doesn't disable IABR, it just re-uses the padding for both v6 `armv6m` and `armv8m`. I believe this is the correct padding. This is the ARM CMSIS definition for armv8m NVIC that has this padding

https://github.com/ARM-software/CMSIS_5/blob/a5743627a233e02388e583fbcfab29573b8fe643/CMSIS/Core/Include/core_armv8mml.h#L478-L481

Reply:
I don't think it adds up right - on armv8 you'd have 16 words of IABR, then 16 words of padding, then 16 words of ITNS, then 32 words of padding, then IPR after that, which is 80 words in total instead of the correct 64.

For clarity I'd much rather have something like this, where every item is present or replaced by a same-sized padding:

```rist
    #[cfg(not(armv6m))]
    pub iabr: [RO<u32>; 16],
    #[cfg(armv6m)]
    _reserved4: [u32; 16],

    _reserved5: [u32; 16],

    #[cfg(armv8m))]
    pub itns: [RW<u32>; 16],
    #[cfg(not(armv8m)))]
    _reserved6: [u32; 16],

    _reserved7: [u32; 16],

    #[cfg(not(armv6m))]
    pub ipr: [RW<u8>; 496],
    #[cfg(armv6m)]
    pub ipr: [RW<u32>; 8],
```

Arguably we should write it as `cfg(any(armv7m, armv8m))` instead of `cfg(not(armv6m))`, though it might make it a bit trickier if we one day need to add something like armv9m.

Reply:
Yeah you are right that it is off by 16 words of padding. I was confused by the fact that in the ARMv7 manual, IABR is specified as being 8 words long, but here it uses the ARMv7 register length of 16 words (presumably to support both platforms). I agree your method above would be cleaner. I'll change my PR to use that method.

Reply:
Sorry for leaving this hanging, I just pushed a commit fixing the issue. LMK if there are any other problems